backends/advanced/src/advanced_omi_backend/database.py

```
@@ -386,6 +386,73 @@ async def update_memory_processing_status(
             logger.info(f"Updated memory processing status to {status} for {audio_uuid}")
         return result.modified_count > 0
 
+    async def update_transcription_status(
+        self, audio_uuid: str, status: str, error_message: str = None, provider: str = None
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)